File: src/types/reimbursement.ts
import type { InvoiceCategory } from './invoice'

export interface CategorySummary {
  category: InvoiceCategory
  count: number
  total_amount: number
}

export interface TransportDetail {
  label: string
  count: number
  amount: number
}

export interface HotelLevelDetail {
  level: string
  persons: number
  days: number
  daily_rate: number
  amount: number
  actual_amount: number
}

export interface MealSubsidyDetail {
  persons: number
  days: number
  daily_rate: number
  amount: number
}

export interface ReimbursementForm {
  name: string
  department: string
  destination: string
  travel_start: string
  travel_end: string
  travel_days: number
  companions: number
  transport_details: TransportDetail[]
  transport_subtotal: number
  city_transport_count: number
  city_transport_amount: number
  city_transport_actual_amount: number
  hotel_levels: HotelLevelDetail[]
  hotel_subtotal: number
  meal_subsidy: MealSubsidyDetail
  baggage_amount: number
  meal_reimbursement: number
  summaries: CategorySummary[]
  total_amount: number
}

File: src-tauri/src/models/reimbursement.rs
use super::invoice::InvoiceCategory;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategorySummary {
    pub category: InvoiceCategory,
    pub count: usize,      // 单据张数
    pub total_amount: f64, // 申报金额
}

/// 住宿费分级明细
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HotelLevelDetail {
    pub level: String,      // 级别名称 (战区级以上/军级/师级/其他人员)
    pub persons: usize,     // 人数
    pub days: usize,        // 天数（住宿晚数）
    pub daily_rate: f64,    // 每日标准
    pub amount: f64,        // 可报销金额（封顶后）
    pub actual_amount: f64, // 实际支付金额
}

/// 伙食补助明细
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MealSubsidyDetail {
    pub persons: usize,  // 人数
    pub days: usize,     // 天数
    pub daily_rate: f64, // 每日标准
    pub amount: f64,     // 申报金额
}

/// 城市间交通费明细
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TransportDetail {
    pub label: String, // 类别名 (车船票/飞机票/订退改签票)
    pub count: usize,  // 单据张数
    pub amount: f64,   // 申报金额
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReimbursementForm {
    pub name: String,         // 姓名
    pub department: String,   // 部职别
    pub destination: String,  // 到达地点
    pub travel_start: String, // 出差开始日期
    pub travel_end: String,   // 出差结束日期
    pub travel_days: usize,   // 出差天数
    pub companions: usize,    // 同行人数

    // 城市间交通费
    pub transport_details: Vec<TransportDetail>,
    pub transport_subtotal: f64,

    // 市内交通费
    pub city_transport_count: usize,
    pub city_transport_amount: f64,        // 可报销金额（封顶后）
    pub city_transport_actual_amount: f64, // 实际支出金额
    pub city_transport_daily_std: f64,     // 每日标准（元/天），供前端展示与超标分析

    // 住宿费
    pub hotel_levels: Vec<HotelLevelDetail>,
    pub hotel_subtotal: f64,

    // 伙食补助
    pub meal_subsidy: MealSubsidyDetail,

    // 其他
    pub baggage_amount: f64,     // 行李托运费
    pub meal_reimbursement: f64, // 凭据报销伙食费

    // 兼容旧接口
    pub summaries: Vec<CategorySummary>,
    pub total_amount: f64, // 申报金额合计
}
